pkg/setting/setting_test.go

```diff
@@ -189,5 +191,29 @@ func TestLoadingSettings(t *testing.T) {
 			So(cfg.RendererCallbackUrl, ShouldEqual, "http://myserver/renderer/")
 		})
 
+		Convey("session config", func() {
+			Convey("Reading session should log error ", func() {
+				var (
+					called = false
+					msg    = ""
+					cfg    = NewCfg()
+				)
+
+				monkey.Patch(log.Warn, func(testMessage string, v ...interface{}) {
```

Comment:
Cool lib. Didn't know you could do that in Go :)

That's said I dont want monkey patching in Grafana. Its very un Go like and forgetting to unpatch something could result in endless hours of debugging why something breaks somewhere else. In this case a custom mock that checks if Warn have been called is the preferred solution. But to be honest I dont think we need to test for log calls either.

Reply:
Pretty interesting lib. I like the guy is honest about it :D:
> Monkey is not threadsafe. Or any kind of safe.
> // this function is super unsafe
// aww yeah
// It copies a slice to a raw memory location, disabling all memory protection before doing so.

Also the repo does not have issues enabled which does not boost confidence. @markelog did you use this lib before?

Reply:
It would be possible to test with a mock instead since `log.Logger` is an interface. The code would be almost the same except without using unsafe code :)

Reply:
Please refer to the license: https://github.com/bouk/monkey/blob/master/LICENSE.md

Don’t use the project, it’s satirical 😁